cli/src/commands/auth.rs

```diff
@@ -0,0 +1,50 @@
+use crate::commands::Command;
+use crate::config::CliConfig;
+use crate::errors::{ExitCode, Fallible};
+use crate::style::KEY;
+use crate::telemetry::Session;
+use crate::terminal::sensitive;
+use console::style;
+use log::{debug, error, info, warn};
+use structopt::StructOpt;
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Auth {
+    /// Link the CLI to your Apollo account using your Personal API Key.
+    Setup(Setup),
+}
+
+#[derive(StructOpt)]
+pub struct Setup {}
+
+impl Command for Setup {
+    fn run(&self, session: &mut Session) -> Fallible<ExitCode> {
+        session.log_command("auth setup");
+
+        debug!("Checking is config's api_key is already set...");
+        if session.config.api_key.is_some() {
+            warn!("Authentication already configured.");
+        }
+
+        info!("To link your CLI to your Apollo account go to {} and create a new Personal API Key. Once you've done that, copy the key and paste it into the prompt below.",
+            style("https://engine.apollographql.com/user-settings").cyan());
+        let key_input = sensitive("Personal API Key:")?;
+        let key = key_input.trim();
```

Comment:
FYI: Did you know that in rust you can repurpose a variable name?
```rust
let key = sensitive("Personal API Key:")?;
let key = key.trim();
```

Reply:
🤯 I did not. I think VSCode must have been lagging when I accidentally typed that the first time and it highlighted red.